### ISSUE TITLE
Flash HaxBall, HTML5 HaxBall and "extrapolation"

### ISSUE THREAD

**Issue by 9999OldSchool** [haxball/haxball-issues#1054]
Hello @basro . I get that we won't ever return to Flash HaxBall, it's definitely fine. But we're having huge problem at the moment if not since extrapolation has been added. I get that extrapolation helps some people to reduce the input delay. I had the input delay back then, too. The HTML5 HaxBall is new, we all have been impatient as it is completely understandable. As an oldschool player, I don't think the thing "extrapolation" has been as positive thing. HTML5 HaxBall before the extrapolation did exist, was way too superior to Flash HaxBall, we all enjoyed playing HTML5 back then. But it is cancerous as fuck these days if not already over 2 years. 
People are just spamming these poor balls in corners, without any skills. In these top matches (mostly 4v4 gamemode) you simply can't create chances like you used to be in Flash HaxBall even though I'm still one of the best players who have won anything (I'm not gonna say my name, people will now call me as nonamer as expected). 
These players in Flash were just too trash for this game because it took more skills to be good as us (oldschool players), now they're like us when it comes about skills, because the "extrapolation" is making these players way too easy to make tricks, that's why we all can't create chances. To better explain "chances", I will show you to look goals at some recs (top matches) and please, tell us what do you think about this.
I would appreciate your thoughts, @basro .

a) the title deciding match: https://thehax.pl/forum/powtorki.php?nagranie=cb02beef55922cf42e641effe712bed4 3:55, 8:22, 13:35. These goals are just completely random, of course, it is luck sometimes, but if you've been playing the game, you would know it's not even luck.

b) the final cup match: https://thehax.pl/forum/powtorki.php?nagranie=0dabecffd607cf9788c77c957ae4cff4 11:33, it's just a new meta like these fake kicks (which is even a feature request, even with new meta we can't create chances which is so annoying, you can just pray for these goals to not be in some matches)

c) the HaxBall Champions League final: https://thehax.pl/forum/powtorki.php?nagranie=2d8dadb4b2491a014fa5cfbe3fa839bd 37:11, 39:42, 44:06, new meta = random goals, not luck and that's it.

I can also say that these top players prefer Flash gameplay over HTML gameplay, if it would change your mind or something like that, I will reply without any problems, thanks. 

Is it too hard to put the same physics (same codes) to HTML5 HaxBall from Flash HaxBall? I don't mind on the design, doesn't even matter. We all just want to have the fun even it's dead game, but still. If you liked making the game, then you should listen people playing your game and compare thoughts = more money = more updates. It's just a huge lack of communication between the owner of the game and players. 

#BringTheFlashGameplayBack

**Comment by Jacob-1998** [comment]
Good post and nice to see others raising the same concerns. What you notice is how random and unpredictable the spams/rockets are. This is because the Flash Engine added a tiny Input Delay (please basro find out this number), which gave so much structure and predictability to the game, which is now disappeared completely.

Without structure and predictability, how can top players recognise patterns and use them to their advantage? They cant. The Input Delay from Flash also made out of position players punished more for their mistakes, the jukes where much more exaggerated and fun, and higher skill ceiling, 0 downside to this Input Delay that the Flash Machine added, only positives.

This is what Saviola, who has a good understanding of game engines, told me:
'Adding a synthetic delay should be easy, convincing basro that it is good might be the harder part.'

Please basro wake up to whats happening, everybody is complaining about cancer gameplay so much because we all remember how it used to be. We miss Flash Physics and we miss 0 Extrapolation, both of them.

Also heres an interesting thought, if you cant remove Extrapolation but do manage to match the physics to the same as the Flash Machine, maybe the Extrapolation wont be an advantage on those physics anyway. Because the Flash skills rely much more on tempo (extrapolation messes with your timing), not who has the quickest possible fingers and reaction times. I think if you fixed the physics to match Flash, extrapolation wouldn't even be a problem anymore, only something legitimate laggers would use, instead of people looking for advantages!

**Comment by Meyer511** [comment]
I agree with Jacob the old gameplay in flash was much more fun and skillful. I know you couldnt fix extrapolation but this is a good idea to make the flash input delay on html, because then extrapolation wouldnt help anymore and extrapolation abusers would disappear, you fix both problems of bad gameplay and extrapolation abuse with 1 update.

**Comment by 9999OldSchool** [comment]
@Jacob-1998 I completely agree with you. Every mistake in Flash HaxBall was punishment, which we took it seriously and always played safe how it should be. If basro listened us back then, this game would have been one of the most played football games. This game had so much potential to become even more popular than League Of Legends and CS:GO, we've all seen it in YeS league back then. 

@Meyer511 Yes, that's what I said. The problem is that basro doesn't even play the game, to understand the game. If he listened to us, it would be way better and more fun. Sadly, but maybe it's still not late!

**Comment by 9999OldSchool** [comment]
@Jacob-1998 I completely agree with you. Every mistake in Flash HaxBall was punishment, which we all took is seriously and played safe how it should be. In HTML5 there's not punishment, which causes these random goals (not luck), and doesn't have space for creating useful tactics as it was used to be in Flash HaxBall and chances. 

@Meyer511 Yes, I've said it. The problem is that Basro doesn't play the game (or he actually plays, but not competitively), to understand the game well to understand problems. Sadly, but it's still not late to fix the game, hopefully!

**Comment by iscoman** [comment]
I'm playing this game for years and I couldn't agree more.
today's game is all random, It's all based on spam battles and extrapolation abusers.

I can only guess that you're probably very busy and you rather ignore some of the bugs/problems.
which is completely understood since you own nothing to anyone.
I wish you could find the time to handle these problems and fix them once in for all.

I especially agree with what Jacob mentioned in his post, if it's possible to match the physics of the game to the previous flash version that would be great and maybe it's worth a try?

**Comment by 9999OldSchool** [comment]
@iscoman Yes, it's worth a try. But the problem is that a lot of players will complain, it's a tricky situation though. I'm sure you remember when basro limited extrapolation from 200 to 50? Then all players complained about that, and basro was afraid of it, so he "deleted" the update again. It probably would be same situation, again. Also, when he limited extrapolation, it was smooth, but not enough to match the Flash HaxBall. That's the problem.

**Comment by Jacob-1998** [comment]
Well according to Saviola its very simple to add an artificial Input Delay. Thats all basro needs to do, find the Input Delay on Flash and match that number, artificially add it on HTML.

If he cant find that number, I and many here would be more than happy to test different numbers until we find the one that we agree feels exactly like Flash again.

Yes people will complain like always, but right now so many people complain about HTML anyway. The Flash gameplay stood the test of time and lasted 7 years with many leagues, HTML already after 2 years has only 2 leagues left, and the same players in both leagues not even different teams.

Extrapolation wont be a problem when the skills/tackles on this game are based on timing again.

So basro, PLEASE digest all of this and take action. This is a very SIMPLE solution, that will fix TWO SERIOUS issues that exist - extrapolation + gameplay quality.

**Comment by Jacob-1998** [comment]
This is a good example of the FLASH Heavier Movement (Input Delay) physics.

https://image.ibb.co/gO6pYn/asd.gif

First jasko can intercept the ball cleanly because he made a perfect read on Doktor, instead of a a random spam tackle that cant be controlled.

Now that jasko intercepted with a lift like he intended, he can punish Doktor for being defensively out of position. **With the heavy movement Doktor cant cover both the NORTH and SOUTH side above and below him, without flying.** This is where in HTML he wouldn't be punished for his mistake. On FLASH as you see, he chooses to cover the SOUTH side and misses the ball, he doesn't recover with his heavy movement pulling him away from jasko.

huub is also out of position, so he gets juked the same way as Doktor, chooses SOUTH side and cant also reach NORTH side afterwards like he would with sharper/nimble HTML movement. huub had the chance to stop this goal if he put himself in the right position early enough, but he is punished for not reading the situation quickly (low iq player being punished) and therefore being out of position.


This is just one example of why FLASH Heavy Movement makes for a much more fun game that rewards better players, allows more creativity and cool goals, and inspires lower players to learn from their mistakes and improve (actively engage their brain with the game which is more interactive and addicting) instead of not even knowing they're making mistakes but getting away with them, and then frustrating the good players who can't capitalise from their superior experience/skill/brain/timing etc..

**Comment by mddrr123** [comment]
tbh, they all are right and i agree with all of them. this game was fun to play, there were lots of players, leagues back then. but now? people only come to this game for official games. most of them already retired. It's not fun anymore. It's full of randomly spams and goals now. So basro, plz read our comments and at least try to change something

**Comment by Jacob-1998** [comment]
> This is a good example of the FLASH Heavier Movement (Input Delay) physics.
> 
> https://image.ibb.co/gO6pYn/asd.gif
> 
> First jasko can intercept the ball cleanly because he made a perfect read on Doktor, instead of a a random spam tackle that cant be controlled.
> 
> Now that jasko intercepted with a lift like he intended, he can punish Doktor for being defensively out of position. **With the heavy movement Doktor cant cover both the NORTH and SOUTH side above and below him, without flying.** This is where in HTML he wouldn't be punished for his mistake. On FLASH as you see, he chooses to cover the SOUTH side and misses the ball, he doesn't recover with his heavy movement pulling him away from jasko.
> 
> huub is also out of position, so he gets juked the same way as Doktor, chooses SOUTH side and cant also reach NORTH side afterwards like he would with sharper/nimble HTML movement. huub had the chance to stop this goal if he put himself in the right position early enough, but he is punished for not reading the situation quickly (low iq player being punished) and therefore being out of position.
> 
> This is just one example of why FLASH Heavy Movement makes for a much more fun game that rewards better players, allows more creativity and cool goals, and inspires lower players to learn from their mistakes and improve (actively engage their brain with the game which is more interactive and addicting) instead of not even knowing they're making mistakes but getting away with them, and then frustrating the good players who can't capitalise from their superior experience/skill/brain/timing etc..



ALSO I have to add to this, extrapolation wouldn't save those 2 players from getting juked. In fact, their timing would be off, and they would commit to the tackle too early and just get juked even more possibly. Its just hard to time interceptions, and position yourself at the right times, when using extrapolation.

I really feel this would kill 2 birds with 1 stone so to speak.

**Comment by 9999OldSchool** [comment]
And here we go again.
Sadly, this time about HaxBall Champions League (HCL) final that just happened yesterday.
I feel like we're fighting for "meta" (where I said in my first post, that it's not luck anymore). I was figuring how it's not luck, so I feel like this HTML5 has got Aim Assist. Do you know what it is?
If you're/have playing/ed Fortnite with a controller on PC or Console, you would know how it feels, unfortunanely.

https://www.haxball.com/replay?v=3#https://hax.saviola.de/r/?h=51666653b8e41b8657794a90322c35e7.hbr2

5:35, god bless, just remove the extrapolation. Even Abduction pressed x, but didn't work because of extrapolation, obviously.
21:15, what to say.
45:00, no any hate on Smokin' Aces, well played, guys. I'm just showing these goals to @basro how it feels playing HTML5 HaxBall. 

Sadly that we keep just getting bad moments which is really unfair for all us. This game would have had a lot, lot potential if you were listening to us. No one likes the way how the game is created at the moment, bruh. It's all based on battle spams, which is really "nice"? 
That's why there are always drama on FM 4v4 (only European league 4v4) -> https://www.fm-haxball.co.uk , there we just keep getting more and more protest about macro players, why?
Because it's all based on battle spams, so they install macro and they're good to go, lmfao?

Remove the extrapolation or at least reinstall the game without extrapolation, give Flash physics to HTML5 HaxBall and we'll see how it goes. I'm pretty sure it would be enjoyable (not including these abusers of extrapolation). 
Sadly that is futsal map currently the biggest map on HTML5 HaxBall at the moment, which is expected since it reminds me on good old days. 

Damn, @basro please, at least listen to some player and speak with him about why, how, what, etc. 
Trust me.

Best regards, 
9999OldSchool.

**Comment by kingjasko** [comment]
Agree with most of what's being said in this post. I guess I played this game long enough to have a valid view on the game, so I'll share my view, as well.

First off, I don't think any of us want flash back, you can go ahead and try it here - http://www.haxball.com/flash/ you will quickly realize how bad it actually was. 

But I really think the current gameplay has been ruined for a long time on HTML5. The main reason would be the major advantage that extrapolation gives. Imagine using 0 extrapolation vs 100 extrapolation, bounce the ball on the wall for a few minutes and you'll see the difference. At first 100 extrapolation will seem laggy and useless, but in the long run, the 100ms prediction helps A LOT with the skill, you bounce the ball on the wall, but you see everything 100ms in the future, you can control it way better once your brains adapt to it, do faster tricks, etc. Now imagine having 8 players on the pitch abusing the same thing. Movement is faster, but space is limited, you'll get a robotic gameplay with 0 creativity and 0 options, the only way to score would be through spam and fake kicks, which is what's happening nowadays. The only downside is the jitter and "teleporting" that comes with it, ofc and even that is mostly annoying only when you don't have possession.

Now, I'm not the type to brag about my "achievements", but after helping my team win 2 of the biggest competitions in Haxball recently, Feed Me and Haxball Champions League I believe I know what I'm talking about. 

For me the strategic side of the game interests me the most, so I spent past few seasons developing a tactic that would work best in terms of positioning with and without possession. I basically noticed it's much harder to create something in attack, so with certain positioning you can neutralize even the best players in the game, but I don't want to play like this, no one does, I could be greedy and want it to stay this way, since I found a way to "abuse" it, but it's not enjoyable. Even gameplay on a map that was usually the complete opposite to 4v4 3def , Real Soccer has completely changed after everyone started using extrapolation. Passing, teamwork and creativity used the be the main traits of this map, but recently everything revolves around abusing throws and closing space by blocking. Matches are 20 minutes long, but you'd easily spend 15 minutes with a "dead" ball around throw-in line, fighting for millimeters of space and trying to trick your opponents to get a throw for your own team. It has lost all it's charm. The game became robotic and boring, even if you wanted to play differently, you simply can't, there's no space to do anything if the opponents decide to play like that. It's same for both big and Real Soccer.

So, overall I'd say keep HTML5, keep extrapolation as it obviously helps a lot of people (me included), but the current Haxball is robotic, narrowminded and simply brainless. Something has to be done about it. 

Another big issue would be the fake kick. Basro said it's a feature, but why does the effect increase with more extrapolation? Surely that's not intended nor wanted, right? So a lot of people are using higher extrapolation to get the power of mindfucking opponents with fake kicks and such, since it's impossible to trust your own eyes anymore in this era of Haxball. 1st step in the right direction would be to strip extrapolation of all the advantages, except reducing delays, the reason it was implemented for.

**Comment by Jacob-1998** [comment]
> So, overall I'd say keep HTML5, keep extrapolation as it obviously helps a lot of people (me included), but the current Haxball is robotic, narrowminded and simply brainless. Something has to be done about it.


Yes, only change the physics to match Flash. Then extrapolation wont be such an issue since we will have the heavy movement, and we will have creativity back and less spam gameplay. Also more patterns in the game that can be recognised and used for tactics/creativity/moments of genius.

**Comment by Jacob-1998** [comment]
Just so you know basro we are not all talking out of our arse, making stuff up out of thin air. These are real problems that have gone on for too long, Haxball can do much better than this and I know you can too, its possible. We are just being vocal about it, but so many people hate these problems too, and many have already quit because of them instead of wasting their time complaining.

**Comment by Jacob-1998** [comment]
And basro so you know, kingjasko just won the HCL Final
http://www.youtube.com/watch?v=3hmY5TWERnw&t=48m28s

And FM Div 1 on the last season:
https://i.imgur.com/S3S471s.png

With his team Smokin' Aces.

He just won the 2 hardest competitions on this game, and even he is saying it needs to be changed, even though these physics suit him right now.

**THE PHYSICS ARE SO BAD FOR GAMEPLAY RIGHT NOW, THAT EVEN THE PEOPLE WHO SUCCEED ON IT, WANT IT CHANGED BACK!**

**Comment by basro** [comment]
@Jacob-1998 I'll humour you this one time and take the opportunity to warn you that if you keep spamming on unrelated issues I'll start having to moderate you. And also to tell you that doing such things makes me less likely to want to reply to you.

Nothing in this issue is useful, I've read it all.

From what I gather the things proposed are:
- Remove extrapolation: I already tried this, it was a shit-fest, I wont try it again. I've already explained why it can't be removed. It has nothing to do with me agreeing or disagreeing with it being bad for the game. I don't want to keep repeating myself. I've already said myself that I'd not have added this feature if I knew this would happen.
- Add input lag: Seriously? No, this is dumb and I will not do it.
- Vague make physics "like they were before": This doesn't help me, physics are the same in the code.

Regarding the fake kick, it's a feature in the sense that prediction is a feature. Prediction is not perfect, it makes errors. Fake kicks happen when the prediction was wrong. Using extrapolation amplifies the effect of prediction errors, **but only for your own client**. Your extrapolation settings will have no effect on how others percieve the game. Using high extrapolation values doesn't make your "fake kick" strats more effective.

**Comment by 9999OldSchool** [comment]
I understand that you're the only man who knows everything about Haxball like these stuffs, codes, etc.
Could you tell an answer why extrapolation can't be removed because I don't know the answer, though.  But I remember, back then you decreased the extrapolation between -50 and 50 if I'm not wrong. It was really enjoyable and smooth in only 1 day, sadly. Did you cancel the update because people complained about it or? I would like to know these answers.

Anyways, regarding the fake kick, I don't agree at all with you. If someone's using high extrapolation value, he'll be more effective when he's about to fake kick (easier to fake kick if you know what I mean), but not for someone who is using 20 extrapolation **even if it doesn't have effect for other players**, for example.

Honestly, I think the last update before you added extrapolation update fucked up Haxball. Before that update, it was playable and smooth as fuck, it was way too superior to Flash Haxball. After that, you updated something and almost all players started to have input delay including me, it wasn't even the same physics, I can remember it really well. Dunno what you updated before extrapolation update, I'm pretty sure.


**Nothing in this issue is useful, I've read it all.** by basro. Well, do you know some things that might change the entire Haxball then? Would be nice if you could try some things, though.
It's really painful to play this game at the moment and that there's no similar game like Haxball which is even more painful. ^^

**Comment by kingjasko** [comment]
Well, fake kick should in theory work like you said, but in reality, guys with high (100-200) extrapolation, doing fake kicks or standard movement will appear laggy and jittery to others, even if you're using 0 extra (unplayable with delays, even 30-40 that I'm using now is giving me a fair bit of delay, but extrapolation just ruins the game, even though it gives advantages on top tier of haxball). High extrapolation also makes it much easier to abuse this "feature". If you can't change anything else, at least limit the extrapolation to 100, if 50 was too drastic for that vocal minority that made you change it back to how it is now (it was literally 5 people who were mad about losing their advantage over others, but because they were so loud and threatening you decided you'd better change it back). But, yea, like I said in my previous post, fake kicking on 0 extra vs 100-200 extra is not comparable (it's hard on 0, like it was on flash and even when successful, it won't be as drastic as on higher extrapolation, that's what I'm trying to tell you), you'd know what I'm talking about if you played the game on a high level. You think we're imagining it, because when you try using more extrapolation you just see the bad side of it and you don't properly test it. I'm wasting my time to try and help you improve this game, so it's playable for the next few years, but you refuse our help with some scientific answers. :/

**Comment by Wazarr94** [comment]
Adding a limitation to extrapolation is not effective, as it's very easy to bypass (you can modify the game's code to remove that limit). basro stated that when he restored the 200 limit.

Also, his "scientific" answer is simply a logical answer, the physics code was not changed.

**Comment by Jacob-1998** [comment]
> * Add input lag: Seriously? No, this is dumb and I will not do it.
> * Vague make physics "like they were before": This doesn't help me, physics are the same in the code.

The Flash Engine added tiny Input Delay to all actions, thats why the physics are different despite having the same code.

How can you disregard an idea as stupid, when its been backed by 3 HCL winners in just this topic alone. Would it hurt to consider it, or even let us test a room with it at least?

As you remember I gave you a long detailed email explaining the precise details as to why the TINY Input Delay (arguably 0.05msec or less, needs testing though), gave the game MUCH more structure and depth and fun and addicting everything else already stated.

Is this an issue of you not believing us about the Physics being different? If so, go in a Flash room and move with the ball, you will notice how much 'heavier' the movement feels, from the Tiny Input Delay, this is what we want back. We had enough of pinball Haxball and spam games with this 0 Input Delay HTML Engine physics.

Easy solution to this is add a feature where the hold can make an Input Delay Value from 0.01msec to 0.20msec, and we can test this stuff ourselves and tell you if its better gameplay, and which number gives that same feeling like Flash had.

You are too smart to disregard this without any thought as 'dumb' basro, we can make Haxball better again, its possible.

The physics on Flash were not problematic, annoying or dumb, they were perfect.

Theres a reason Haxball was a huge success on Flash, and regular Haxball was the most popular mode. IDK if you know this basro, but Futsal has practically taken over this game as the most popular game mode now for serious players. Ask yourself, would this happen if the HTML Engine 0 Input Delay physics were good as the Flash Engine Tiny Input Delay physics?

**Comment by basro** [comment]
Using extrapolation with negative values (up to -50) will give you input lag, so you can already test all you want.

In any case, for the same reason extrapolation can't be removed: Artificial input lag can't be added because people will hack the client to remove it. There's no cheat proof way to implement this.

You guys see me as a stubborn developer. You don't realize you are asking for things that are quite literally impossible to implement in a way that isn't trivial to bypass. My hands are tied by reality.

I understand your frustration with me, but it comes from not understanding how things work.

It is entirely possible that flash had slightly higher input lag (although I don't really feel it and many people have complained to me that html5 has higher input lag). It is also a fact that flash had higher latency. These two effects combined could very well change how the hb classic game plays, the higher ping makes some jukes more effective since it's harder to read what your opponent is doing. It is possible that this made the game worse. However, the solution is not to add lag. The solution should be to tweak the game physics to make it harder to defend.

@kingjasko extrapolation is a rendering setting, it only affects how the game is drawn on your screen. If I were to make a a version of haxball that has rendering disabled I'd not even have to read the extrapolation variable. It is impossible for it to affect how other players see the results of your input.

I will proceed to continue reading this issue (I always read, all issues), but I will probably not reply here again unless there's something new for me to say. Please try to understand that I can't spend hours trying to explain things in here. I don't do it out of spite.

**Comment by Jacob-1998** [comment]
As far as I understand, latency is when something has happened in the game, but is concealed and there is a delay between when it happened and when you see it happen.

If this is lower in HTML, thats absolutely fine, much better.

Its about the heavy movement, not the ping or latency or anything else like that. The old heavy movement is the key for this game to be good again, with patterns skills tactics tackles and more all being at a much higher level (mostly thanks to patterns in all those situations, which disappear when theres no Tiny Input Delay).

If you can't add some 0.05msec delay to movement and kicks, then I have another idea/feature request, I'll make a new thread on that, please dont take it as spam since its a new feature request based on what you've just told me, not spamming the same complaints as before.

**Comment by kingjasko** [comment]
Well, basro I love the communication, that's what's the most important thing to me. I've been reaching out to you for a long time now to give you my opinion on how the game feels and compares to flash on the highest level. No hard feelings.

**Comment by AnddyAnddy** [comment]
Just saying that basro is right at least on one point that none of you mentionned

It is waaaaaay easier to juke a player having 100 ping than 30 (the current average of western europe)
And in the years 2011-2015, vps didn't exist in haxball, every room was self made by the player who had the better net, and the ping rarely went under 60
More ping - > having the ball let the owner trust his gameplay whereas opponents trusts their eyes -> easier juke

So maybe (and probably) the gameplay was more enjoyable, but this is harshly due to the ping more than the physic
I can bet that if every single player start using extrapolation -15 and a handicap x where x is 100-player's ping, we'll all get the flash feelings


Oh and the argument of the number of players that reduced is senseless, tell me ONE flash game in 2005/2015 that didnt have its number of players being reduced, and the haxball community being very toxic to new players doesn't help, that has nothing to deal with flash/html5 war
In my opinion, the game is slowly dying but that's normal, who wanna play "flash" game in 2020, the hype is over


Anyway, since some of our coders have already totally deep copied the games, it's a matter of time before we move onto these "new" haxball, maybe to prevent that, your solution would be to put the game as open source and to accept push requests, so you'll keep earning money with ads and haxball will naturally live again.
The real issue of haxball is that you rarely make updates even when some of them are easy to do.

**Comment by 9999OldSchool** [comment]
Sorry for tagging you again, @basro 

Someone posted and it proves, as I said in previous comments, that extrapolation just needs to be removed and that's it. Could you explain why do you not listen to us, though? I don't see any reason to not listen to us, at least try to listen us and try to communicate, it would be appreciated by many players...

I would like you to look at this post: https://fm-haxball.co.uk/community/viewtopic.php?p=91629#p91629

Unlucky, have a good day.

**Comment by kingjasko** [comment]
> 
> 
> Sorry for tagging you again, @basro
> 
> Someone posted and it proves, as I said in previous comments, that extrapolation just needs to be removed and that's it. Could you explain why do you not listen to us, though? I don't see any reason to not listen to us, at least try to listen us and try to communicate, it would be appreciated by many players...
> 
> I would like you to look at this post: https://fm-haxball.co.uk/community/viewtopic.php?p=91629#p91629
> 
> Unlucky, have a good day.

Something is brewing, stay patient :)

**Comment by besthxplayer** [comment]
Hi @basro, In my opinion the game works well, I hope you don't remove the extrapolation, since very few people will like it. Most of them are used to playing with extraolation and there are several new players who are coming to haxball thanks to tik tok for example and it is easy for them to get used to playing with high extrapolation, they have no complaints.
I think you can easily predict x fakes if you know when they are going to use them, I used 200 extrapolation many months ago and I have no input lag currently.
ignore few people, most of players are used to it and have no problem using extrapolation.
regards
